x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import {
+  EuiForm,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiSelect,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiButton,
+  EuiPanel,
+} from '@elastic/eui';
+import { useActions, useValues } from 'kea';
+
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { FlashMessages } from '../../../shared/flash_messages';
+import {
+  ALLOWED_CHARS_NOTE,
+  ENGINE_CREATION_FORM_ENGINE_LANGUAGE_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER,
+  ENGINE_CREATION_FORM_SUBMIT_BUTTON_LABEL,
+  ENGINE_CREATION_FORM_TITLE,
+  ENGINE_CREATION_TITLE,
+  SANITIZED_NAME_NOTE,
+  SUPPORTED_LANGUAGES,
+} from './constants';
+import { EngineCreationLogic } from './engine_creation_logic';
+
+export const EngineCreation: React.FC = () => {
+  const { name, rawName, language } = useValues(EngineCreationLogic);
+  const { setLanguage, setRawName, submitEngine } = useActions(EngineCreationLogic);
+
+  return (
+    <div data-test-subj="EngineCreation">
+      <SetPageChrome trail={[ENGINE_CREATION_TITLE]} />
+      <EuiPageHeader>
+        <EuiPageHeaderSection>
+          <EuiTitle size="l">
+            <h1>{ENGINE_CREATION_TITLE}</h1>
+          </EuiTitle>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <FlashMessages />
+      <EuiPanel>
```

Comment:
I _think_ this should be [EuiPageBody](https://elastic.github.io/eui/#/layout/page), correct me if I'm wrong

```suggestion
      <EuiPageBody>
```

I wonder if we should put the FlashMessages inside or outside the page body as well - I think I've been putting them inside so far but we may need to check w/ Davey for consistency going forward